### ISSUE TITLE
Webview and other views cannot remember their height

### ISSUE THREAD

**Issue by tamuratak** [microsoft/vscode#106585]
Version: 1.50.0-insider
Commit: cca20eba8974b1d6b6a9b71bf4e60a29c8f26648
Date: 2020-09-12T00:56:22.923Z
Electron: 9.3.0
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0
OS: Darwin x64 17.7.0

I have tested the Webview View API. That greatly works. However, Webview views cannot remember their height. Expanding and collapsing them change their total height. How can extension authors work around it?

Steps to Reproduce:

1. Expanding and collapsing Webview views and other views.
2. Their total height gets much larger.

![Sep-13-2020 20-20-49](https://user-images.githubusercontent.com/10665499/93016863-3ac9f100-f5ff-11ea-89db-4106f219e35d.gif)

**Comment by alexr00** [comment]
@mjbvz this makes GHPRI look pretty bad as the webview-view ends up taking up most of the space: https://github.com/microsoft/vscode-pull-request-github/issues/3227.

**Comment by mjbvz** [comment]
@alexr00 Are you still seeing this in the GitHub PR extension? I just tested but it seems like the view takes up half of the space

![Screen Shot 2022-02-11 at 4 43 23 PM](https://user-images.githubusercontent.com/12821956/153689090-6ebda03a-24a6-4754-9c99-89b4318dec26.png)
 
If you still see this, can you please provide specific steps to reproduce this?

**Comment by mjbvz** [comment]
Thanks. I can't yet reproduce this on desktop but can hit this on web following those steps

However I also reproduce this with the outline view if I move it into the `Github PR` view container so I don't think it's webview view specific

@alexr00 / @sbatten   Do you have any pointers on how initial view sizes are computed?

**Comment by sbatten** [comment]
@mjbvz this is where we expand the views and set size. https://github.com/microsoft/vscode/blob/4a130c40ed876644ed8af2943809d08221375408/src/vs/base/browser/ui/splitview/paneview.ts#L141-L167

**Comment by mjbvz** [comment]
@alexr00 Is there an easy way to reproduce this with a local build? This is a complete pain to debug on GitHub.dev and I don't know how to get into this state on local desktop/web

**Comment by tamuratak** [comment]
Version: 1.66.0-insider (Universal)
Commit: 6dca4c1a1f7bb3529074601eb32938526fa9ecc7
Date: 2022-03-04T05:18:46.542Z
Electron: 13.5.2
Chromium: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Darwin arm64 20.6.0

I still see the original issue on VS Code (Desktop). Notice the order of expanding. When reloading github.dev, I guess restoring views would be restored in the same order by chance, which causes the issue.

https://user-images.githubusercontent.com/10665499/156739508-2a4beb98-6c3f-4d2c-9a89-4db1872d6fe1.mov

**Comment by mjbvz** [comment]
@tamuratak Thanks!

@sbatten Adding you since I can reproduce this using any types of views. I tested using the source control views that git lens adds which are all normal tree views, not webview views

**Comment by mjbvz** [comment]
I talked with @sbatten about this and we determined the original issue is by-design.

The [most recent demo video](https://github.com/microsoft/vscode/issues/106585#issuecomment-1059007241) shows why this is the case:

1. The lower view is collapsed and remembers its height
1. At this point the upper view now has a larger height to till in the space left open by the upper view
1. The upper view is then collapsed and also remembers its current height
1. We now start expanding the views again. First the lower view is expanded to its original height (from step 1)
1. Then the upper view is expanded to the height from (step 3)

This results in the overall height of both views increasing but both views were correctly preserving their height through all of the stages

We don't think there's a good way to fix this. The example above is a minimal repo, but there are many other ways that users could get into this situation with more complex sequences of actions

So I'm closing this specific issue as by-design but will open a new issue to continue investigating the issue with the GitHub pr extension seeing views not restored the the expected heights on first load / reload
